src/character.cpp

```diff
@@ -1910,55 +1910,25 @@ bool Character::is_blind() const
 
 bool Character::pour_into( item &container, item &liquid )
 {
-    if( liquid.is_ammo() && ( container.is_tool() || container.is_gun() ) ) {
-        // TODO: merge this part with game::reload
-        // for filling up chainsaws, jackhammers and flamethrowers
-
-        if( container.ammo_type() != liquid.ammo_type() ) {
-            add_msg_if_player( m_info, _( "Your %1$s won't hold %2$s." ), container.tname().c_str(),
-                               liquid.tname().c_str() );
-            return false;
-        }
-
-        if( container.ammo_remaining() >= container.ammo_capacity() ) {
-            add_msg_if_player( m_info, _( "Your %1$s can't hold any more %2$s." ), container.tname().c_str(),
-                               liquid.tname().c_str() );
-            return false;
-        }
-
-        if( container.ammo_remaining() && container.ammo_current() != liquid.typeId() ) {
-            add_msg_if_player( m_info, _( "You can't mix loads in your %s." ), container.tname().c_str() );
-            return false;
-        }
+    std::string err;
+
+    const bool allow_bucket = &container == &weapon || !has_item( container );
```

Comment:
It's a [pre-existing hack](https://github.com/CleverRaven/Cataclysm-DDA/pull/17348/files#diff-f1933acfcc0056a11de5cb2ad6fd6caaL1946).